tiledb/array.py

```diff
@@ -163,6 +148,21 @@ def index_domain_subarray(array, dom, idx: tuple):
 
         start, stop, step = dim_slice.start, dim_slice.stop, dim_slice.step
 
+        # In the case that current domain is non-empty, we need to consider it
+        if (
+            start is None
+            and stop is None
+            and hasattr(array.schema, "current_domain")
```

Comment:
When is this line false?

Reply:
This is true when no explicit index is requested (i.e., `A[:]`) and the array has a non-empty current domain. In all other cases, this is false.